backend/cfg/cfgize.ml

```diff
@@ -459,6 +415,32 @@ let rec add_blocks :
       | Cfg.Tailcall _ | Cfg.Call_no_return _ ->
         body
     in
+    let can_raise =
+      (* Recompute [can_raise] and check that instruction in the middle of the
+         block do not raise, i.e., only the terminator, or the last instruction
+         (when the terminator is just a goto) can raise. *)
+      let last_can_raise =
+        match (terminator.Cfg.desc : Cfg.terminator) with
+        | Always _ -> true
+        | Raise _
+        | Tailcall (Func _)
+        | Call_no_return _ | Never | Parity_test _ | Truth_test _ | Float_test _
+        | Int_test _ | Switch _ | Return
+        | Tailcall (Self _) ->
+          false
+      in
+      let rec check = function
+        | [] -> false
+        | [last] ->
+          let res = Cfg.can_raise_basic last.Cfg.desc in
+          assert ((not res) || last_can_raise);
+          res
+        | hd :: tail ->
+          assert (not (Cfg.can_raise_basic hd.Cfg.desc));
+          check tail
+      in
+      Cfg.can_raise_terminator terminator.Cfg.desc || check body
```

Comment:
That does no feel right: if the terminator can raise,
we will not check that nothing in the body can raise.

Reply:
fixed, thank you!